### PR TITLE
Replace transport spec path with cloudevents

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan [myan@redhat.com](mailto:myan@redhat.com)

### CODE REVIEW

File: agent/pkg/status/controller/managedclusters/clusters_status_sync.go [hunk 1]
```diff
@@ -34,10 +34,11 @@ func AddClustersStatusController(mgr ctrl.Manager, producer producer.Producer, l
 		})
 	}
 
-	predicateFunc := func() bool { // bundle predicate
-		return hubOfHubsConfig.Data["aggregationLevel"] == "full" ||
-			hubOfHubsConfig.Data["aggregationLevel"] == "minimal"
+	predicateFunc := func() bool {
+		// return hubOfHubsConfig.Data["aggregationLevel"] == "full" ||
+		// 	hubOfHubsConfig.Data["aggregationLevel"] == "minimal"
 		// at this point send all managed clusters even if aggregation level is minimal
```
Comment (clyang82) [inline agent/pkg/status/controller/managedclusters/clusters_status_sync.go:40]: Can we continue supporting this configuration?

File: pkg/transport/consumer/generic_consumer.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	var receiver interface{}
+	switch transportConfig.TransportType {
+	case string(transport.Kafka):
+		var err error
+		log.Info("transport consumer with kafka receiver")
+		receiver, err = protocol.NewKafkaReceiver(transportConfig.KafkaConfig)
+		if err != nil {
+			return nil, err
+		}
+	case string(transport.Chan):
```
Comment (clyang82) [inline pkg/transport/consumer/generic_consumer.go:38]: Does it for test only?

File: pkg/transport/producer/generic_producer.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+type GenericProducer struct {
+	log              logr.Logger
+	client           cloudevents.Client
+	messageSizeLimit int
+}
+
+func NewGenericProducer(transportConfig *transport.TransportConfig) (*GenericProducer, error) {
+	var sender interface{}
+	messageSize := 960 * 1000
```
Comment (clyang82) [inline pkg/transport/producer/generic_producer.go:31]: Can we make it configurable?

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: pkg/transport/consumer/generic_consumer.go (hunk 2 above)
Comment (yanmxa) [reply pkg/transport/consumer/generic_consumer.go:38]: Yeah, this is for unit test.

File: pkg/transport/producer/generic_producer.go (hunk 3 above)
Comment (yanmxa) [reply pkg/transport/producer/generic_producer.go:31]: It is configurable. the` messageSize := 960 * 1000` is the default value. Actually, it reads the config value from the below line `messageSize = transportConfig.KafkaConfig.ProducerConfig.MessageSizeLimitKB * 1000`

File: agent/pkg/status/controller/managedclusters/clusters_status_sync.go (hunk 1 above)
Comment (yanmxa) [reply agent/pkg/status/controller/managedclusters/clusters_status_sync.go:40]: The `aggregationLevel` is only for policies, it should return `true` for the managed cluster

**Pull request merged**